src/frontend/change_buffer.rs

```diff
@@ -64,12 +68,25 @@ pub enum RevisionSelection {
     Range(Revision, Revision),
 }
 
+fn resolve_command(config: &Config, key: char) -> Option<Command> {
+    config
+        .keys
+        .change
+        .get(&key)
+        .map(|key_def| {
+            let KeyDefinition::Command(command) = key_def;
+            command
+        })
+        .copied()
+}
+
 fn read_keys(
     mut ev_selection: EventWriter<ChangeBufferSelectionEvent>,
     mut change_buffer: Query<&mut ChangeBuffer>,
     mut ev_keypresses: EventReader<KeyEvent>,
     mut exit: EventWriter<AppExit>,
     mut navigation: Navigation,
+    config: Res<Config>,

```

Comment:
This isn't bad. I think a good next step is to extract this into its own event mapper (so that instead of listening to `EventReader<KeyEvent>` we can listen to e.g. `EventReader<Command>`.